integration/run_test.go

```diff
@@ -520,3 +521,152 @@ func TestRunInNamespace(t *testing.T) {
+			},
+		},
+	}
+
+	tests := []struct {
+		name           string
+		subUIDContents string
+		subGIDContents string
+		checkFiles     []checker
+	}{
+		{
+			name:           "with one set of substitutions",
+			subUIDContents: fmt.Sprintf("%s:12345:1001", dummyuser),
+			subGIDContents: fmt.Sprintf("%s:12345:1001", dummyuser),
+			checkFiles: []checker{
+				{
+					path:        "/usr/bin/sh",
+					expectedUID: "12345",
+					expectedGID: "12345",
+				},
+			},
+		},
+		{
+			name:           "with multiple substitutions",
+			subUIDContents: fmt.Sprintf("%s:12345:1000\n%s:22222:1", dummyuser, dummyuser),
+			subGIDContents: fmt.Sprintf("%s:12345:1000\n%s:22222:1", dummyuser, dummyuser),
+			checkFiles: []checker{
+				{
+					path:        "/usr/bin/sh",
+					expectedUID: "12345",
+					expectedGID: "12345",
+				},
+			},
+		},
+	}
+
+	for _, mode := range modes {
+		mode := mode
+		for _, tt := range tests {
+			tt := tt
+			t.Run(tt.name+" "+mode.name, func(t *testing.T) {
+				regConfig := newRegistryConfig()
+				sh, done := newShellWithRegistry(t, regConfig)
+				defer done()
+
+				sh.X("groupadd", "-g", "12345", dummygroup)
+				sh.X("useradd", "-u", "12345", "-g", "12345", "-m", dummyuser)
+
+				sh.Pipe(nil, shell.C("echo", tt.subUIDContents), shell.C("tee", uidPath))
+				sh.Pipe(nil, shell.C("echo", tt.subGIDContents), shell.C("tee", gidPath))
+
+				rebootContainerd(t, sh, getContainerdConfigToml(t, false), getSnapshotterConfigToml(t, false))
+				imageInfo := dockerhub(imageName)
+				sh.X("nerdctl", "pull", "-q", imageName)
+
+				filenames, err := sh.OLog("ls", baseSnapshotDir)
+				if err != nil {
+					t.Fatalf("error listing files in %s", baseSnapshotDir)
+				}
+
+				// Copy image, remove blobs, and re-pull with SOCI
+				copyImage(sh, dockerhub(imageName), regConfig.mirror(imageName))
+				indexDigest := mode.indexBuilderFn(sh, regConfig.mirror(imageName))
+				if indexDigest != "" {
+					sh.X("soci", "push", "--user", regConfig.creds(), regConfig.mirror(imageName).ref)
+				}
+				sh.X("rm", "-rf", filepath.Join(store.DefaultSociContentStorePath, "blobs", "sha256"))
+
+				pullCmd := imagePullCmd
+				if indexDigest != "" {
+					pullCmd = append(pullCmd, "--soci-index-digest", indexDigest)
+				}
+				sh.X(append(pullCmd, regConfig.mirror(imageName).ref)...)
+				// time.Sleep(999999999999999999)
+				containerID := strings.TrimSpace(string(sh.O("nerdctl-with-idmapping", "run", "-d",
+					"--net", "none",
+					"--pull", "never",
+					"--userns", dummyuser,
+					"--snapshotter", "soci",
+					imageInfo.ref, "sleep", "infinity",
+				)))
+
+				newFilenames, err := sh.OLog("ls", baseSnapshotDir)
+				if err != nil {
+					t.Fatalf("error listing files in %s", baseSnapshotDir)
+				}
+
+				if len(filenames) == len(newFilenames) {
+					t.Fatalf("error: id-mapping failed")
+				}
+
+				for _, check := range tt.checkFiles {
+					fullCheckPath := filepath.Join(baseRuntimeDir, containerID, "rootfs", check.path)
```

Comment:
is this checking each file uid gid to match the expected uid gid?

Reply:
In each of the specified files to check, yes